internal/runbits/requirements/requirements.go

```diff
@@ -81,22 +92,34 @@ func executePackageOperation(prime primeable, packageName, packageVersion string
 		language, err := model.LanguageByCommit(pj.CommitUUID())
 		if err == nil {
 			langName = language.Name
-			ns = model.NewNamespacePkgOrBundle(langName, nsType)
+			if nsType == model.NamespacePackage || nsType == model.NamespaceBundle {
+				ns = model.NewNamespacePkgOrBundle(langName, nsType)
+			}
+		} else {
+			logging.Error("Could not get language from project: %v", err)
 		}
+
 		out.Notice(locale.Tl("operating_message", "", pj.NamespaceString(), pj.Dir()))
 	}
 
-	var validatePkg = operation == model.OperationAdded
+	switch nsType {
+	case model.NamespaceLanguage:
+		ns = model.NewNamespaceLanguage()
+	case model.NamespacePlatform:
+		ns = model.NewNamespacePlatform()
+	}
+
```

Comment:
This feels like it should be more closely tied to the logic happening on line 96. Perhaps this switch case is where you look up the language if encountering a package / bundle.